resources/js/Components/Galleries/GalleryPage/GalleryControls.tsx

```diff
@@ -65,19 +61,17 @@ export const GalleryControls = ({
                             icon="Heart"
                             className={cn(hasLiked && "button-like-selected")}
                             onClick={() => {
-                                if (!authenticated) {
-                                    showConnectOverlay(() => {
-                                        void like(gallery.slug, true);
-
-                                        router.reload({
-                                            only: ["stats", "gallery"],
-                                        });
-                                    });
+                                signedAction(({ authenticated }) => {
+                                    // If user wasnt authenticated, foce a positive
+                                    // like since we dont know if he liked it before
```

Comment:
```suggestion
                                    // If user wasnt authenticated, force a positive
                                    // like since we dont know if he liked it before
```

typo, but also: can this cause weird scenarios where you had it liked before and click the button to unlike and get prompted to sign and afterwards it's still liked? I guess the assumption here is that if you're logged out your intention is to like it as there is no prior state for it? 🤔 might be good enough

Reply:
@ItsANameToo this logic only applies on cases that you are not logged-in since we dont know if the gallery was liked for you or not, In that, as I user, I would expect a "up-like"  

If you are logged but not signed, it toggles as usual after signed